src/twyn/core/config_handler.py

```diff
@@ -1,54 +1,147 @@
 import logging
+from dataclasses import asdict, dataclass
+from enum import Enum
 from os import getcwd
 from pathlib import Path
 from typing import Any, Optional
 
-from tomlkit import dumps, parse
+from tomlkit import TOMLDocument, dumps, parse, table
 
-from twyn.base.constants import DEFAULT_PROJECT_TOML_FILE
+from twyn.base.constants import (
+    DEFAULT_PROJECT_TOML_FILE,
+    DEFAULT_SELECTOR_METHOD,
+    AvailableLoggingLevels,
+)
 from twyn.core.exceptions import (
     AllowlistPackageAlreadyExistsError,
     AllowlistPackageDoesNotExistError,
+    TOMLError,
 )
 
 logger = logging.getLogger()
 
 
+@dataclass(frozen=True)
+class TwynConfiguration:
+    """Fully resolved configuration for Twyn."""
+
+    dependency_file: Optional[str]
+    selector_method: str
+    logging_level: AvailableLoggingLevels
+    allowlist: set[str]
+
+
+@dataclass(frozen=True)
+class ReadTwynConfiguration:
+    """Configuration for twyn as set by the user. It may have None values."""
+
+    dependency_file: Optional[str]
+    selector_method: Optional[str]
+    logging_level: Optional[AvailableLoggingLevels]
+    allowlist: set[str]
+
+
 class ConfigHandler:
-    """Read certain values into a central ConfigHandler object."""
+    """Manage reading and writing configurations for Twyn."""
 
     def __init__(self, file_path: Optional[str] = None, enforce_file: bool = True):
         self._file_path = file_path or DEFAULT_PROJECT_TOML_FILE
         self._enforce_file = enforce_file
-        self._toml = self._get_toml_as_dict()
-        self._twyn_data = self._get_twyn_data()
 
-        self.dependency_file: Optional[str] = self._twyn_data.get("dependency_file")
-        self.selector_method: Optional[str] = self._twyn_data.get("selector_method")
-        self.logging_level: Optional[str] = self._twyn_data.get("logging_level")
-        self.allowlist: set[str] = set(self._twyn_data.get("allowlist", []))
+    def resolve_config(
```

Comment:
I've been thinking for a while, could it be that this class' responsibilities are too broad?

For instance, right now this class both creates mechanisms to create config objects and interacting with the config file, would it make sense to split it?

Say, we have something that acts as a _factory_, just creating config objects, and then we have the config objects themselves, who are the ones that read and write to the file.

Maybe this is not the PR to make these changes, as some methods are mandatory both when creating the object and when interacting with the file (like `_read_toml`) but this could be adapted whenever we move this class' implementation to use the FileHandler.

Reply:
Overall agree that it would not hurt to separate the reading and the writing. Though I imagine they will need mechanisms to share stuff.

Having said that, this class has 3 public methods, and 150 lines or so. It's not too bad. I am not a fan of the config objects _writing themselves_ though.

I do not really know how useful the filehandler will be for this, but I guess I'll see in the next PRs.